android/src/main/java/com/epicshaggy/biometric/NativeBiometric.java

```diff
@@ -156,408 +151,455 @@ public void isAvailable(PluginCall call) {
-                byte[] signatureBytes = signature.sign();
-
-                signatureResult = Base64.encodeToString(signatureBytes, Base64.DEFAULT);
-                ret.put("signedData", signatureResult);
-                call.resolve(ret);
-            } catch (UserNotAuthenticatedException e) {
-                //Exception thrown when the user has not been authenticated
-                call.reject("User has not been authenticated");
-            } catch (KeyPermanentlyInvalidatedException e) {
-                //Exception thrown when the key has been invalidated for example when lock screen has been disabled.
-                call.reject("Process interrupted");
-            } catch (Exception e) {
-                throw new RuntimeException(e);
+                getKeyStore().deleteEntry(KEY_ALIAS);
+                SharedPreferences.Editor editor = getContext().getSharedPreferences(NATIVE_BIOMETRIC_SHARED_PREFERENCES, Context.MODE_PRIVATE).edit();
+                editor.clear();
+                editor.apply();
+                call.resolve();
+            } catch (KeyStoreException e) {
+                call.reject("Failed to delete", e);
+            } catch (CertificateException e) {
+                call.reject("Failed to delete", e);
+            } catch (NoSuchAlgorithmException e) {
+                call.reject("Failed to delete", e);
+            } catch (IOException e) {
+                call.reject("Failed to delete", e);
             }
+        } else {
+            call.reject("No server name was provided");
         }
+    }
 
-        @PluginMethod()
-        public void deleteCredentials ( final PluginCall call){
-            String KEY_ALIAS = call.getString("server", DEFAULT_KEY);
+    @RequiresApi(api = Build.VERSION_CODES.N)
+    @PluginMethod()
+    public void getPublicKey(final PluginCall call) throws
+            GeneralSecurityException, IOException {
+        JSObject ret = new JSObject();
 
-            if (KEY_ALIAS != null) {
-                try {
-                    getKeyStore().deleteEntry(KEY_ALIAS);
-                    SharedPreferences.Editor editor = getContext().getSharedPreferences(NATIVE_BIOMETRIC_SHARED_PREFERENCES, Context.MODE_PRIVATE).edit();
-                    editor.clear();
-                    editor.apply();
-                    call.resolve();
-                } catch (KeyStoreException e) {
-                    call.reject("Failed to delete", e);
-                } catch (CertificateException e) {
-                    call.reject("Failed to delete", e);
-                } catch (NoSuchAlgorithmException e) {
-                    call.reject("Failed to delete", e);
-                } catch (IOException e) {
-                    call.reject("Failed to delete", e);
-                }
-            } else {
-                call.reject("No server name was provided");
-            }
+        if(doesBiometricKeyExist()) {
+            deleteBiometricKey();
         }
```

Comment:
Question, why if it's exists we delete the existing biometric key when we try to fetch public key?

Reply:
I guess the user is only supposed to fetch the public key one time only when in enrollment, so I guess it's better to delete the previous key. The usage for verification not needed the use of public key, instead the key pair itself.

Reply:
Allright, I guess that's acceptable since public key will also overwrited everytime you enroll. I agree wtih this behaviour